test/unit-tests/inputField.test.js

```diff
@@ -0,0 +1,115 @@
+let { openBrowser, goto, inputField, closeBrowser, intervalSecs, timeoutSecs, below } = require('../../lib/taiko');
+let { createHtml, openBrowserArgs, removeFile } = require('./test-util');
+const test_name = 'inputfield';
+
+describe('inputField', () => {
+
+    beforeAll(async () => {
+        await openBrowser(openBrowserArgs);
+    }, 30000);
+
+    afterAll(async () => {
+        await closeBrowser();
+    }, 30000);
+
+    describe('with inline text', () => {
+        let filePath;
+        beforeAll(() => {
+            let innerHtml = '<form> \n' +
+                '<input type="text" name="user[login]" id="user[login]" /> \n' +
+                '<input type="text" name="user[email]" id="user[email]" /> \n' +
+                '<input type="password" name="user[password]" id="user[password]" /> \n' +
+                '</form>';
+            filePath = createHtml(innerHtml, test_name);
+        });
+
+        beforeEach(async () => {
+            await goto(filePath);
+        });
+
+        afterAll(() => {
+            removeFile(filePath);
+        });
+
+        test('test exists()', async () => {
+            await expect(inputField({ id: 'user[login]' }).exists()).resolves.toBeTruthy();
+            await expect(inputField({ id: 'user[email]' }).exists()).resolves.toBeTruthy();
+            await expect(inputField({ id: 'user[re-password]' }).exists(intervalSecs(0), timeoutSecs(0))).resolves.toBeFalsy();
+        });
+    });
+
+    describe('wrapped in label', () => {
+        let filePath;
+        beforeAll(() => {
+            let innerHtml = '<form> \n' +
+                '<label><input type="text" name="user[login]" id="user[login]" /></label> \n' +
+                '<label><input type="text" name="user[email]" id="user[email]" /></label> \n' +
+                '<label><input type="password" name="user[password]" id="user[password]" /></label> \n' +
+                '</form>';
+            filePath = createHtml(innerHtml, test_name);
+        });
+
+        beforeEach(async () => {
+            await goto(filePath);
+        });
+
+        afterAll(() => {
+            removeFile(filePath);
+        });
+
+        test('test exists()', async () => {
+            await expect(inputField({ id: 'user[login]' }).exists()).resolves.toBeTruthy();
```

Comment:
this test can be updated to use wrapped in label instead of using attr and value pair